src/main/java/com/saralein/server/connection/ConnectionHandler.java

```diff
@@ -0,0 +1,27 @@
+package com.saralein.server.connection;
+
+import com.saralein.server.logger.ILogger;
+import com.saralein.server.response.ResponseHandler;
+
+import java.io.IOException;
+
+public class ConnectionHandler implements Runnable {
+    private final ILogger logger;
+    private final Connection socket;
+
+    public ConnectionHandler(Connection socket, ILogger logger) {
+        this.socket = socket;
+        this.logger = logger;
+    }
+
+    public void run() {
+        synchronized (this) {
+            try {
+                ResponseHandler.sendResponse(socket);
```

Comment:
Should a `ConnectionHandler` have a dependency on the `ResponseHandler`? It feels like these abstractions are at varying levels of abstraction.

Reply:
Also, I would like rephrase this
>It feels like these abstractions are at varying levels of abstraction.

What I am trying to say is that `ResponseBuilder` feels like it is at a higher abstraction level than the `ConnectionHandler`, but `ConnectionHandler` depends on `ResponseBuilder`.

Reply:
I changed it to have `ConnectionHandler` depend on a `Response` interface (which `ResponseBuilder` implements), which is then injected into `ConnectionHandler`.  [See here.](https://github.com/saralein/java-server/pull/1/commits/33e748c9040443de81ae24115b8ff94d24064f0a)  It feels like this will need to be updated a bit in the future.